### PR TITLE
Adding unit type back / fixes on pretty printing.

### PR DESCRIPTION
**Pull request by rodrigogribeiro**
* Adding the unit type back, which solve some ambiguities which are present when treating this type as 0-tuple.
* Fixes function pretty printing, which was wrongly dealing with type variables and constraints.

### CODE REVIEW

File: src/Solcore/Frontend/Pretty/SolcorePretty.hs [hunk 1]
```diff
@@ -316,8 +321,12 @@ instance Pretty Ty where
     = ppr t1 <+> (text "->") <+> ppr t2
   ppr (TyCon n ts)
     | isTuple n = parens $ commaSep (map ppr ts)
+    | isUnit n = text "()"
```
Comment (mbenke) [inline src/Solcore/Frontend/Pretty/SolcorePretty.hs:324]: How should unit be spelled in source files? "unit" or "()"?

**Review by mbenke — CHANGES_REQUESTED** [review]
It would be better with a (simple) test

**Review by mbenke — APPROVED** [review]

**Comment by mbenke** [comment]
What about parsing?

also please see that test/examples/cases/Ref.solc  is updated

**Comment by rodrigogribeiro** [comment]
Fixed in the last commit.

File: src/Solcore/Frontend/Pretty/SolcorePretty.hs (hunk 1 above)
Comment (rodrigogribeiro) [reply src/Solcore/Frontend/Pretty/SolcorePretty.hs:324]: Fixed. Notation is `()`.

**Pull request merged**